netrc.go

```diff
@@ -114,26 +138,55 @@ func (netrc Netrc) Save() error {
 	writer := bufio.NewWriter(file)
 
 	for key, value := range netrc.Entries {
-		if key == "" {
-			writer.WriteString("default\n")
+		// If the entry has been parsed there is a
+		// print format string that we should use
+		if (value.PrintFormat != "") {
+			values := []interface{}{key}
+			// Append the values that are not empty
+			if (value.Account != "") {
+				values = append(values, value.Account)
+			}
+			if (value.Login != "") {
+				values = append(values, value.Login)
+			}
+			if (value.Password != "") {
+				values = append(values, value.Password)
+			}
+			// Generate string from format and values
+			result := fmt.Sprintf(value.PrintFormat, values...)
+			writer.WriteString(result)
 		} else {
-			writer.WriteString("machine " + key + "\n")
-		}
+			if key == "" {
+				writer.WriteString("default\n")
+			} else {
+				writer.WriteString("machine " + key + "\n")
+			}
 
-		if value.Account != "" {
-			writer.WriteString("\taccount " + value.Account + "\n")
-		}
-		if value.Login != "" {
-			writer.WriteString("\tlogin " + value.Login + "\n")
-		}
-		if value.Password != "" {
-			writer.WriteString("\tpassword " + value.Password + "\n")
+			if value.Account != "" {
+				writer.WriteString("\taccount " + value.Account + "\n")
+			}
+			if value.Login != "" {
+				writer.WriteString("\tlogin " + value.Login + "\n")
+			}
+			if value.Password != "" {
+				writer.WriteString("\tpassword " + value.Password + "\n")
+			}
 		}
-
-		writer.WriteString("\n")
 	}
 
 	writer.Flush()
 
 	return nil
 }
+
+func main() {
+	var netrc, _ = Parse()
+
+	var entry *Entry = nil
```

Comment:
About package main..i did that just for quick testing and as i still need
to do the other problem i left it as is.
On Jul 30, 2015 18:45, "Lorenz Leutgeb" <notifications@github.com> wrote:

> In netrc.go <https://github.com/coduno/netrc/pull/1#discussion_r35885329>:
>
> >  	}
> >
> >  	writer.Flush()
> >
> >  	return nil
> >  }
> > +
> > +func main() {
> > +	var netrc, _ = Parse()
> > +
> > +	var entry *Entry = nil
>
> Should be:
>
> entry := new(Entry)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coduno/netrc/pull/1/files#r35885329>.
>